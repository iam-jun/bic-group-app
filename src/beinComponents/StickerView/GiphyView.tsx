import {
  GiphyContent,
  GiphyGridView,
  GiphyMedia,
  GiphyMediaView,
} from '@giphy/react-native-sdk';
import React, {useState} from 'react';
import {SafeAreaView, ScrollView} from 'react-native';
import SearchInput from '../inputs/SearchInput';

interface Props {
  height?: number;
}

export default function GiphyView({height}: Props) {
  const [searchQuery, setSearchQuery] = useState<string>('');
  const [media, setMedia] = useState<GiphyMedia | null>(null);

  const request = searchQuery
    ? GiphyContent.search({searchQuery: searchQuery})
    : GiphyContent.trending({mediaType: undefined});

  return (
    <SafeAreaView>
      <SearchInput
        onChangeText={setSearchQuery}
        placeholder="Search..."
        value={searchQuery}
      />
      <GiphyGridView
        content={request}
        cellPadding={3}
        style={{height: '100%', marginTop: 16}}
        onMediaSelect={e => setMedia(e.nativeEvent.media)}
      />
      {media && (
        <ScrollView
          style={{
            aspectRatio: media.aspectRatio,
            maxHeight: 400,
            padding: 24,
            width: '100%',
          }}>
          <GiphyMediaView
            media={media}
            style={{aspectRatio: media.aspectRatio}}
          />
        </ScrollView>
      )}
    </SafeAreaView>
  );
}
