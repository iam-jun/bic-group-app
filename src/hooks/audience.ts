import {useSelector} from 'react-redux';
import {IObject} from '~/interfaces/common';

const useAudience = () => {
  const audiences = useSelector((state: IObject<any>) => state.audience);
  return audiences;
};

export default useAudience;
