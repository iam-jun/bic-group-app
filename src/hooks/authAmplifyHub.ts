import {useEffect} from 'react';
import {Auth, Hub} from "aws-amplify";
import {useDispatch} from "react-redux";
import {HubCapsule} from "@aws-amplify/core/src/Hub";

import * as actions from '~/screens/Auth/redux/actions';

const useAuthAmplifyHub = () => {
    const dispatch = useDispatch();

    useEffect(() => {
        const handleAuthEvent = ({payload: {event, data}}: HubCapsule) => {
            switch (event) {
                case "signIn":
                    Auth.currentAuthenticatedUser()
                        .then(userResponse => {
                            dispatch(actions.signInSuccess(userResponse));
                        })
                        .catch((error) => {
                            console.log('\x1b[36m', 'namanh --- SignInError |  : ', error, '\x1b[0m');
                        });
                    break;
                case "signOut":
                    break;
                case "customOAuthState":
                    break;
            }
        };
        Hub.listen('auth', handleAuthEvent);
        return () => {
            Hub.remove('auth', handleAuthEvent);
        };
    }, []);
}

export default useAuthAmplifyHub;
