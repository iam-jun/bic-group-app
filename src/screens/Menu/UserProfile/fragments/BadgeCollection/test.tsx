import React from 'react';
import {
  View, Dimensions, FlatList, StyleSheet,
} from 'react-native';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import Avatar from '~/baseComponents/Avatar';
import Icon from '~/baseComponents/Icon';
import spacing, { borderRadius } from '~/theme/spacing';

const Grid = ({ data }: {data: any[]}) => {
  const deviceWidth = Dimensions.get('window').width;

  const itemWidth = 48; // Constant item width
  const numColumns = Math.floor((deviceWidth - 10 * 2) / (itemWidth + 6 * 2));
  const rowWidth = numColumns * (itemWidth + 6 * 2);
  const paddingLeft = (deviceWidth - (10 * 2) - rowWidth) / 2;

  const theme: ExtendedTheme = useTheme();
  const { colors } = theme;
  const styles = themeStyles(theme);

  const isSelected = false;

  const renderGridItem = ({ item, index }: any) =>
    // Render each grid item here
    (
      <View
        key={index}
        style={{
          margin: 6,
          borderWidth: isSelected ? 2 : 1,
          borderColor: isSelected ? colors.purple50 : colors.neutral5,
          borderRadius: borderRadius.pill,
        }}
      >
        <Avatar.Medium
          isRounded
          source={{ uri: item?.iconUrl }}
        />
        {isSelected
          ? (
            <View style={styles.iconChose}>
              <Icon size={10} icon="Check" tintColor={colors.white} />
            </View>
          )
          : null}
      </View>
    );

  return (
    <View style={{
      flex: 1,
      justifyContent: 'center',
      alignItems: 'center',
      paddingHorizontal: 10,
      paddingBottom: 10,
    }}
    >

      <View style={{
        // paddingLeft,
      }}
      >
        <FlatList
          data={data}
          renderItem={renderGridItem}
          keyExtractor={(item, index) => index.toString()}
          numColumns={numColumns}
        />
      </View>
    </View>
  );
};

const themeStyles = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      flex: 1,
      paddingTop: spacing.padding.large,
    },
    iconChose: {
      backgroundColor: colors.purple50,
      borderRadius: borderRadius.pill,
      height: 20,
      width: 20,
      alignItems: 'center',
      justifyContent: 'center',
      position: 'absolute',
      zIndex: 1,
      right: 0,
    },
  });
};

export default Grid;
