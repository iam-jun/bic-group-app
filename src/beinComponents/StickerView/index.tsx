import {GiphyContent, GiphyGridView, GiphyMedia} from '@giphy/react-native-sdk';
import i18next from 'i18next';
import {useKeyboard} from '@react-native-community/hooks';
import React, {useEffect, useImperativeHandle, useMemo, useRef} from 'react';
import {
  DeviceEventEmitter,
  Keyboard,
  NativeSyntheticEvent,
  Platform,
  StyleSheet,
  View,
} from 'react-native';
import {
  Directions,
  FlingGestureHandler,
  FlingGestureHandlerStateChangeEvent,
  State,
} from 'react-native-gesture-handler';
import {useTheme} from 'react-native-paper';
import Animated, {
  runOnJS,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';
import {useSafeAreaInsets} from 'react-native-safe-area-context';
import {dimension} from '~/theme';
import {ITheme} from '~/theme/interfaces';
import {EmojiBoardProps} from '../emoji/EmojiBoard';
import SearchInput from '../inputs/SearchInput';
import LoadingIndicator from '../LoadingIndicator';

export interface Props extends Partial<EmojiBoardProps> {
  stickerViewRef: any;
  onMediaSelect: (media: GiphyMedia) => void;
}

const SEARCH_BAR_HEIGHT = 80;

const _StickerView = ({stickerViewRef, onMediaSelect}: Props) => {
  const INITIAL_KEYBOARD_HEIGHT = 336;
  const [visible, setVisible] = React.useState(false);
  const [isExpanded, setIsExpanded] = React.useState(false);
  const [keyboardHeight, setKeyboardHeight] = React.useState(
    INITIAL_KEYBOARD_HEIGHT,
  );
  const [searchQuery, setSearchQuery] = React.useState('');
  const [loading, setLoading] = React.useState(true);
  const _stickerViewRef = stickerViewRef || useRef();

  const request = searchQuery
    ? GiphyContent.search({searchQuery: searchQuery})
    : GiphyContent.trending({mediaType: undefined});

  const height = useSharedValue(0);

  const theme = useTheme() as ITheme;
  const styles = createStyle(theme);
  const insets = useSafeAreaInsets();

  const keyboard = useKeyboard();

  const calculateContentHeight = useMemo(() => {
    const offset = dimension.headerHeight + insets.top + keyboardHeight;

    return dimension.deviceHeight - offset;
  }, [keyboardHeight]);

  const calculateFullHeight = useMemo(() => {
    const offset = dimension.headerHeight + insets.top;

    return dimension.deviceHeight - offset;
  }, [dimension.deviceHeight]);

  const animatedStyle = useAnimatedStyle(() => {
    return {
      // height: interpolate(
      //   height.value,
      //   [0, keyboardHeight + SEARCH_BAR_HEIGHT],
      //   [0, keyboardHeight + SEARCH_BAR_HEIGHT],
      // ),
      height: height.value,
    };
  });

  useEffect(() => {
    if (
      keyboard?.keyboardHeight &&
      keyboardHeight !== keyboard?.keyboardHeight
    ) {
      setKeyboardHeight(keyboard?.keyboardHeight);
    }
    onKeyboardVisibleChanged(keyboard?.keyboardShown);
  }, [keyboard?.keyboardShown]);

  useEffect(() => {
    DeviceEventEmitter.emit('sticker-board-visible-change', visible);
  }, [visible]);

  const show = () => {
    if (visible) return;

    setVisible(true);

    Keyboard.dismiss();

    height.value = withTiming(keyboardHeight, {
      duration: 200,
    });
  };

  const expand = () => {
    const onDone = () => {
      setIsExpanded(true);
    };

    const expandHeight = keyboard.keyboardShown
      ? calculateContentHeight
      : calculateFullHeight;

    height.value = withTiming(expandHeight, {duration: 200}, () => {
      runOnJS(onDone)();
    });
  };

  const collapse = () => {
    setIsExpanded(false);

    const onDone = () => {
      Keyboard.dismiss();
    };
    height.value = withTiming(
      keyboardHeight + SEARCH_BAR_HEIGHT,
      {
        duration: 200,
      },
      () => {
        runOnJS(onDone)();
      },
    );
  };

  const hide = () => {
    const onHideDone = () => {
      setSearchQuery('');
      setIsExpanded(false);
      setVisible(false);
    };
    height.value = withTiming(0, {duration: 200}, () => {
      runOnJS(onHideDone)();
    });
  };

  const hideImmediately = () => {
    setVisible(false);
    runOnJS(hide)();
  };

  const handleDown = () => {
    if (isExpanded) runOnJS(collapse)();
    else runOnJS(hide)();
  };

  const handleUp = () => {
    runOnJS(expand)();
  };

  const getVisible = () => {
    return visible;
  };

  const onKeyboardVisibleChanged = (keyboardShown: boolean) => {
    if (visible && !keyboardShown) {
      //Wait for keyboard to be invisible
      setTimeout(() => {
        runOnJS(collapse)();
      }, 100);
    } else if (keyboardShown) {
      if (visible) handleUp();
    }
  };

  useImperativeHandle(_stickerViewRef, () => ({
    show,
    hide,
    hideImmediately,
    onBackPress: handleDown,
    getVisible,
  }));

  const onDownFlingHandlerStateChange = ({
    nativeEvent,
  }: FlingGestureHandlerStateChangeEvent) => {
    if (nativeEvent.oldState === State.ACTIVE) {
      runOnJS(handleDown)();
    }
  };

  const onUpFlingHandlerStateChange = ({
    nativeEvent,
  }: FlingGestureHandlerStateChangeEvent) => {
    if (nativeEvent.oldState === State.ACTIVE) {
      runOnJS(handleUp)();
    }
  };

  const onChangeText = (value: string) => {
    setLoading(true);
    setSearchQuery(value);
  };

  const onContentUpdate = () => {
    setLoading(false);
  };

  const _onMediaSelect = (
    e: NativeSyntheticEvent<{
      media: GiphyMedia;
    }>,
  ) => {
    onMediaSelect(e.nativeEvent.media);
  };

  if (!visible) return null;

  const bottomPosition = keyboard.keyboardShown
    ? keyboardHeight + SEARCH_BAR_HEIGHT
    : 0;

  return (
    <FlingGestureHandler
      direction={Directions.DOWN}
      // onEnded={handleDown}
      onHandlerStateChange={onDownFlingHandlerStateChange}>
      <Animated.View
        testID="sticker_view"
        style={[
          styles.container,
          animatedStyle,
          isExpanded && {
            ...styles.expanded,
            // height: calculateContentHeight,
            bottom: Platform.OS === 'android' ? 0 : bottomPosition,
          },
        ]}>
        <View style={styles.stickerView}>
          <FlingGestureHandler
            direction={Directions.UP}
            // onEnded={handleUp}
            onHandlerStateChange={onUpFlingHandlerStateChange}>
            <View style={styles.header}>
              <View style={styles.indicator} />
              <SearchInput
                testID="sticker_view.search_input"
                placeholder={i18next.t('post:comment:search_giphy')}
                value={searchQuery}
                onChangeText={onChangeText}
              />
            </View>
          </FlingGestureHandler>
          <View>
            {loading && <LoadingIndicator style={styles.loading} />}
            <GiphyGridView
              testID="sticker_view.grid_view"
              content={request}
              cellPadding={4}
              style={styles.gridView}
              onContentUpdate={onContentUpdate}
              onMediaSelect={_onMediaSelect}
            />
          </View>
        </View>
      </Animated.View>
    </FlingGestureHandler>
  );
};

const createStyle = (theme: ITheme) => {
  const {colors, spacing} = theme;

  return StyleSheet.create({
    container: {
      backgroundColor: colors.background,
    },
    expanded: {
      position: 'absolute',
      left: 0,
      right: 0,
      zIndex: 99,
    },
    header: {
      paddingVertical: spacing.margin.base,
      marginHorizontal: spacing.margin.base,
      height: SEARCH_BAR_HEIGHT,
    },
    indicator: {
      backgroundColor: colors.borderDivider,
      width: 48,
      height: 4,
      borderRadius: 4,
      alignSelf: 'center',
      marginBottom: spacing.margin.base,
    },
    stickerView: {
      borderTopWidth: 1,
      borderTopColor: colors.borderDivider,
      height: '100%',
    },
    gridView: {
      height: '100%',
    },
    loading: {
      marginTop: spacing.margin.base,
    },
  });
};

const StickerView = React.memo(_StickerView);
// StickerView.whyDidYouRender = true;
export default StickerView;
