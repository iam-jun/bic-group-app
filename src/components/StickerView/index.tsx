import { useKeyboard } from '@react-native-community/hooks';
import { ExtendedTheme, useTheme } from '@react-navigation/native';
import { debounce } from 'lodash';
import React, {
  useCallback, useEffect, useImperativeHandle, useRef,
} from 'react';
import {
  Keyboard, Platform, StyleSheet, View,
} from 'react-native';
import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

import { Portal } from 'react-native-portalize';
import { AppConfig } from '~/configs';
import { dimension } from '~/theme';
import spacing from '~/theme/spacing';
import EmojiSectionIcons from '~/baseComponents/EmojiPicker/components/EmojiSectionIcons';
import EmojiPicker from '~/baseComponents/EmojiPicker';
import useEmojiPickerStore from '~/baseComponents/EmojiPicker/store';
import IEmojiPickerState from '~/baseComponents/EmojiPicker/store/Interface';
import { SearchInput } from '~/baseComponents/Input';
import { useBaseHook } from '~/hooks';
import GiphyView from '../GiphyView';
import { IGiphy } from '~/interfaces/IGiphy';
import SlideUpModal from '../SlideUpModal';

export interface Props {
  stickerViewRef: any;
  onGiphySelected?: (media: IGiphy) => void;
  onEmojiSelected?: (emoji: string) => void;

}

const _StickerView = ({ stickerViewRef, onGiphySelected, onEmojiSelected }: Props) => {
  const { t } = useBaseHook();
  const INITIAL_KEYBOARD_HEIGHT = 336;
  const modalizeRef = useRef<any>();

  const [type, setType] = React.useState('emoji');
  const [visible, setVisible] = React.useState(false);
  const [keyboardHeight, setKeyboardHeight] = React.useState(INITIAL_KEYBOARD_HEIGHT);
  const [searchQuery, setSearchQuery] = React.useState('');
  const _stickerViewRef = stickerViewRef || useRef();
  const emojiPickerRef: any = useRef();

  const actions = useEmojiPickerStore((state: IEmojiPickerState) => state.actions);

  const height = useSharedValue(0);

  const theme: ExtendedTheme = useTheme();
  const styles = createStyle(theme);

  const keyboard = useKeyboard();

  useEffect(
    () => {
      if (keyboard?.keyboardHeight
      && keyboardHeight !== keyboard?.keyboardHeight
      && keyboard?.keyboardHeight > INITIAL_KEYBOARD_HEIGHT
      ) {
        setKeyboardHeight(keyboard?.keyboardHeight);
      }
    }, [keyboard?.keyboardHeight],
  );

  const animatedStyle = useAnimatedStyle(() => ({
    height: height.value,
  }));

  const show = (type:'gif'|'emoji') => {
    setType(type);
    modalizeRef.current?.show();
    Keyboard.dismiss();
  };

  const hide = () => {
    modalizeRef.current?.hide();
  };

  const onChangeText = debounce(
    (value: string) => {
      if (type === 'giphy') {
        setSearchQuery(value);
      } else {
        actions.search(value);
      }
    }, AppConfig.searchTriggerTime,
  );

  const onGiphyPress = (item: IGiphy) => {
    onGiphySelected(item);
  };

  const onEmojiPress = useCallback((emojiName: string) => {
    onEmojiSelected(emojiName);
  }, []);

  const onOpened = () => {
    setVisible(true);

    height.value = withTiming(
      keyboardHeight, { duration: 400 },
    );
  };

  const onClosed = () => {
    setVisible(false);
    setSearchQuery('');

    height.value = withTiming(
      0, { duration: 200 },
    );
  };

  const onCollapsed = () => Keyboard.dismiss();

  const onSearchFocus = () => {
    if (Platform.OS === 'android') return;

    modalizeRef.current?.maximise();
  };

  const onSearchBlur = () => {
    //
  };

  const onBackPress = () => {
    hide();
    return true;
  };

  useImperativeHandle(
    _stickerViewRef, () => ({
      show,
      hide,
      onBackPress,
    }),
  );

  const scrollToSectionIndex = (index: number) => {
    emojiPickerRef.current?.scrollToSectionIndex(index);
  };

  let content = null;

  if (type === 'giphy') {
    content = <GiphyView searchQuery={searchQuery} onSelected={onGiphyPress} />;
  } else if (type === 'emoji') {
    content = (
      <View style={styles.emojiView}>
        <EmojiPicker
          emojiPickerRef={emojiPickerRef}
          onEmojiPress={onEmojiPress}
        />
        <EmojiSectionIcons
          visible={visible}
          onPress={scrollToSectionIndex}
        />
      </View>
    );
  }

  const expandHeight = Platform.OS === 'ios' ? keyboardHeight : undefined;
  const maxHeight = Platform.OS === 'android' ? keyboardHeight : undefined;
  const topOffset = Platform.OS === 'ios' ? dimension.headerHeight : undefined;

  return (
    <View testID="sticker_view" style={styles.container}>
      <Animated.View style={animatedStyle} />
      <Portal>
        <SlideUpModal
          sheetRef={modalizeRef}
          topOffset={topOffset}
          maxHeight={maxHeight}
          expandHeight={expandHeight}
          onOpened={onOpened}
          onClosed={onClosed}
          onCollapsed={onCollapsed}
        >
          <View style={[styles.stickerView]}>
            <View style={styles.header}>
              <SearchInput
                testID="sticker_view.search_input"
                placeholder={t(`sticker:search_${type}`)}
                value={searchQuery}
                onChangeText={onChangeText}
                onFocus={onSearchFocus}
                onBlur={onSearchBlur}
              />
            </View>
            {content}
          </View>
        </SlideUpModal>
      </Portal>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;

  return StyleSheet.create({
    container: {
      backgroundColor: colors.white,
    },
    stickerView: {
      height: '100%',
    },
    emojiView: {
      flex: 1,
    },
    header: {
      paddingVertical: spacing.margin.base,
      marginHorizontal: spacing.margin.base,
    },
    gridView: {
      height: '100%',
    },
    loading: {
      marginTop: spacing.margin.base,
    },
  });
};

const StickerView = React.memo(_StickerView);
StickerView.whyDidYouRender = true;
export default StickerView;
