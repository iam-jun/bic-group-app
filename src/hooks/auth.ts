import {useDispatch, useSelector} from 'react-redux';
import {IObject} from '~/interfaces/common';
import {useEffect} from 'react';
import {Auth} from 'aws-amplify';
import authActions from '~/screens/Auth/redux/actions';
import {put} from 'redux-saga/effects';
import modalActions from '~/store/modal/actions';
import {useBaseHook} from '~/hooks/index';

const useAuth = () => {
  return useSelector((state: IObject<any>) => state.auth);
};

export const useUserIdAuth = () => {
  return (
    useSelector(
      (state: IObject<any>) =>
        state.auth?.user?.signInUserSession?.idToken?.payload?.[
          'custom:bein_user_id'
        ],
    ) || ''
  );
};

export const useAuthToken = () => {
  return (
    useSelector(
      (state: IObject<any>) =>
        state.auth?.user?.signInUserSession?.idToken?.jwtToken,
    ) || ''
  );
};

export const useAuthTokenExpire = () => {
  return (
    useSelector(
      (state: IObject<any>) =>
        state.auth?.user?.signInUserSession?.idToken?.payload?.exp,
    ) || ''
  );
};

export const useAuthKickOut = () => {
  const dispatch = useDispatch();
  const userId = useUserIdAuth();
  const {t} = useBaseHook();

  const checkAuthKickOut = async () => {
    try {
      await Auth.currentAuthenticatedUser({
        bypassCache: true,
      });
    } catch (e) {
      if (!userId) {
        return;
      }
      dispatch(authActions.signOut());
      dispatch(
        modalActions.showAlert({
          title: t('auth:text_kickout_title'),
          content: t('auth:text_kickout_desc'),
          onConfirm: () => put(modalActions.hideAlert()),
          confirmLabel: t('auth:text_kickout_confirm_button'),
        }),
      );
    }
  };

  useEffect(() => {
    checkAuthKickOut();
  }, []);
};

export default useAuth;
