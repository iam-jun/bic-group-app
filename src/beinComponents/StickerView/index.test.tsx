/* eslint-disable @typescript-eslint/no-var-requires */
import * as RNCommunityHooks from '@react-native-community/hooks';
import * as React from 'react';
import * as Animated from 'react-native-reanimated';
import {render, setHookTestState} from '~/test/testUtils';
import StickerView from '.';

describe('StickerView component', () => {
  const stickerViewRef = jest.spyOn(React, 'useRef').mockReturnValue({
    current: {
      show: jest.fn(),
    },
  });
  const baseProps = {
    stickerViewRef,
    onMediaSelect: jest.fn(),
  };

  const reactMock = require('react');
  let Platform = {OS: 'ios'};

  beforeEach(() => {
    jest.spyOn(RNCommunityHooks, 'useKeyboard').mockImplementation(() => {
      return {
        keyboardShown: false,
        coordinates: {
          start: undefined,
          end: {screenX: 0, screenY: 0, width: 0, height: 0},
        },
        keyboardHeight: 0,
      };
    });

    Platform = require('react-native').Platform;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', async () => {
    reactMock.useState = setHookTestState({
      visible: true,
      isExpanded: false,
      keyboardHeight: 0,
      searchQuery: '',
    });
    const rendered = render(<StickerView {...baseProps} />);

    expect(rendered).toMatchSnapshot();
  });

  it('renders null', async () => {
    reactMock.useState = setHookTestState({
      visible: false,
      isExpanded: false,
      keyboardHeight: 0,
      searchQuery: '',
    });
    const rendered = render(<StickerView {...baseProps} />);

    expect(rendered).toMatchSnapshot();
  });

  it(`animated_view should be showed`, () => {
    reactMock.useState = setHookTestState({
      visible: true,
      isExpanded: false,
      keyboardHeight: 0,
      searchQuery: '',
    });

    jest.spyOn(Animated, 'useAnimatedStyle').mockImplementation(() => {
      return {
        height: 336,
      };
    });

    const rendered = render(<StickerView {...baseProps} />);

    const animated_view = rendered.getByTestId('sticker_view.animated_view');

    expect(animated_view).toBeDefined();
    expect(animated_view.props.style.height).toBe(336);
  });

  it(`animated_view should be expanded`, () => {
    reactMock.useState = setHookTestState({
      visible: true,
      isExpanded: true,
      keyboardHeight: 0,
      searchQuery: '',
    });

    const rendered = render(<StickerView {...baseProps} />);

    const animated_view = rendered.getByTestId('sticker_view.animated_view');

    expect(animated_view).toBeDefined();
    expect(animated_view.props.style.height).toBe('100%');
  });

  it('should show KeyboardSpacer if platform is android', () => {
    reactMock.useState = setHookTestState({
      visible: true,
      isExpanded: false,
      keyboardHeight: 0,
      searchQuery: '',
    });

    Platform.OS = 'android';

    const rendered = render(<StickerView {...baseProps} />);

    const keyboard_spacer = rendered.getByTestId(
      'sticker_view.keyboard_spacer',
    );

    expect(keyboard_spacer).toBeDefined();
  });
});
