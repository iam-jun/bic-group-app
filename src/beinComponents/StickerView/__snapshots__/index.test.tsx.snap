// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickerView component renders correctly 1`] = `
<View
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Array [
      Object {
        "height": 1274,
      },
      false,
    ]
  }
  testID="sticker_view"
>
  <View
    style={[Function]}
    testID="sticker_view.animated_view"
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderTopColor": "#EAEDF2",
          "borderTopWidth": 1,
          "height": "100%",
        }
      }
    >
      <View
        collapsable={false}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        style={
          Object {
            "marginHorizontal": 12,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#EAEDF2",
              "borderRadius": 4,
              "height": 4,
              "marginBottom": 12,
              "width": 48,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#EAEDF2",
              "borderRadius": 20,
              "height": 40,
              "justifyContent": "center",
              "paddingHorizontal": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              }
            }
          >
            <View
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={false}
              hitSlop={
                Object {
                  "bottom": 10,
                  "left": 10,
                  "right": 10,
                  "top": 10,
                }
              }
              nativeID="animatedComponent"
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    Object {
                      "marginRight": 8,
                    },
                    Object {
                      "backgroundColor": undefined,
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      undefined,
                      undefined,
                      undefined,
                    ]
                  }
                >
                  <View
                    className="icon-wrapper"
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={20}
                      bbWidth={20}
                      fill="#627286"
                      focusable={false}
                      height={20}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {},
                          Object {
                            "flex": 0,
                            "height": 20,
                            "width": 20,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={20}
                    >
                      <RNSVGGroup
                        fill={4284641926}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          d="M21.71,20.29,18,16.61A9,9,0,1,0,16.61,18l3.68,3.68a1,1,0,0,0,1.42,0A1,1,0,0,0,21.71,20.29ZM11,18a7,7,0,1,1,7-7A7,7,0,0,1,11,18Z"
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
            <TextInput
              autoComplete="off"
              onChangeText={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Search GIPHY"
              placeholderTextColor="#627286"
              returnKeyType="search"
              selectionColor="#627286"
              style={
                Object {
                  "color": "#2F3945",
                  "flex": 1,
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                  "height": "100%",
                }
              }
              testID="sticker_view.search_input"
              value=""
            />
          </View>
        </View>
      </View>
      <GiphyReactNativeGridView
        cellPadding={4}
        content={
          Object {
            "mediaType": undefined,
            "requestType": "trending",
            "searchQuery": "",
          }
        }
        onMediaSelect={[Function]}
        style={
          Object {
            "height": "100%",
          }
        }
        testID="sticker_view.grid_view"
      />
    </View>
  </View>
</View>
`;

exports[`StickerView component renders null 1`] = `null`;
